typescript/introduction/tsc-basic-types/tsc-object-type-comparison.md

```diff
@@ -0,0 +1,173 @@
+---
+author: nem035
+type: normal
+category: must-know
+aspects:
+  - workout
+  - deep
+links:
+  - '[object Type](https://www.typescriptlang.org/docs/handbook/basic-types.html#object){documentation}'
+  - '[Play with object types](https://www.typescriptlang.org/play/index.html#code/MYewdgzgLgBAhgLhgeQEYCsCmxYF4YDeMmYA1gJZJQBOArpjAL4DcAUK6JLKkiBtnkLEylGDXpM2HcNBjAkREhSSoQIADaY4YJjHyKRVOgxZA){website}'
+
+---
+
+# `Object` vs `object` vs `{}`
+
+---
+## Content
+
+TypeScript has 3 ways to represent an object with slightly different behaviors.
+
+> 💡 An object can be thought of as a collection of name value pairs
+
+### `Object`
+
+Since any defined value in JavaScript is an object[1], the `Object` type can be used to represent any type that is considered to contain a value:
+
+```ts
+let num: Object = 3;
+let str: Object = 'enki';
+let arr: Object = [1, 2, 3];
+// ...
+// only null and undefined
+//  give an error
+let n: Object = null; // error
+let u: Object = undefined; // error
+```
+
+This often leads to confusion.
+
+Usually by "object" we only mean collections of named values (e.g. maps, sets, arrays, plain objects), but not primitive values[2] (e.g. numbers or strings).
+
+### `object`
+
+The recommended type to represent an object is `object`.
+
+```ts
+let arr: object = [1, 2, 3];
+let obj: object = { enki: true };
+let map: object = new Map([
+  ["enki", "cool"],
+]);
+// ...
+// primitive values give an error
+let str: object = "enki"; // error
+let num: object = 3; // error
+let n: object = null; // error
+let u: object = undefined; // error
+```
+
+### `{}`
```

Comment:
I think this needs to be surfaced at the top as it's the best way to define objects with concrete shapes. unless you work with some generics, why even use `object`? as it doesn't guard against anything
we also only say using `object` is the recommended way to represent objects

Reply:
reading this insight I get that I should use `object` most of the time, not this